src/constructors.jl

```diff
@@ -19,58 +19,32 @@ function sort_sections(chn::Chains)
 end
 
 """
+    Array(chains[, sections = :parameters;
+          append_chains = true, remove_missing_union = true])
 
-# Array
+Construct an `Array` from a chain.
 
-Array constructor from a Chains object. Returns 3 dimensionsal
-array or an Array of 2 dimensional Arrays. If only a single parameter is selected for
-inclusion, a dimension is dropped in both cases, as is e.g. required by cde(), etc.
+Returns 3 dimensionsal array or an Array of 2 dimensional Arrays. If only a single parameter
+is selected for inclusion, a dimension is dropped in both cases, as is e.g. required by
+cde(), etc.
 
-### Method
-```julia
-  Array(
-    chn::Chains,
-    sections::Vector{Symbol};
-    append_chains::Bool,
-    remove_missing_union::Bool
-  )
-```
+# Examples
 
-### Required arguments
-```julia
-* `chn` : Chains object to convert to an Array
-```
-
-### Optional arguments
-```julia
-* `sections = Symbol[]` : Sections from the Chains object to be included
-* `append_chains = true`  : Append chains into a single column
-* `remove_missing_union = true`  : Convert Union{Missing, Real} to Float64
-```
-
-### Examples
-```julia
 * `Array(chns)` : Array with chain values are appended
 * `Array(chns[:par])` : Array with selected parameter chain values are appended
-* `Array(chns, [:parameters])`  : Array with only :parameter section
+* `Array(chns, :parameters)`  : Array with only :parameter section
 * `Array(chns, [:parameters, :internals])`  : Array includes multiple sections
 * `Array(chns, append_chains=false)`  : Array of Arrays, each chain in its own array
 * `Array(chns, remove_missing_union=false)` : No conversion to remove missing values
-```
-
 """
-function Base.Array(chain::Chains,
-    sections::Union{Symbol, Vector{Symbol}}=Symbol[:parameters];
+function Base.Array(
+    chains::Chains,
+    sections = :parameters;
```

Comment:
It occurs to me that this should probably be `:parameters` if `:parameters in keys(chain.name_map)`, because we can have cases where subsetting gets us a chain without a `:parameter` key.

Reply:
But this problem can occur for any method that allows to specify a `sections` argument that defaults to `:parameters`, can't it? But since we use `_clean_sections` that would not error but just lead to an empty chain in these cases, it seems?

Reply:
True. Should we switch to `nothing` in this case to return the whole chain? Or is it sufficient to return an empty chain? I think returning an empty chain might be confusing if you had done something like `describe(chain[:lp])` or something, where it's clear you only want to see `:lp` but it's locked behind `:internal`. Or maybe this is already happening on the backend.

Reply:
`describe(chain[:lp])` won't work since `chain[:lp]` returns an AxisArray (see https://github.com/TuringLang/MCMCChains.jl/pull/197), but `describe(chain[[:lp]])` would be problematic. I'm not sure what the best (less confusing and most consistent) solution is in this case, maybe one could define
```julia
_default_sections(chains::Chains) = haskey(chains.name_map, :parameters) ? :parameters : nothing
```
and use that as default value for `sections`. Since `name_map` is a NamedTuple that can be inferred as well.

Reply:
That's probably sufficient.